src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,135 @@
+    // instance variables
+    private final WPI_TalonFX flywheelMotor1;
+    private final WPI_TalonFX flywheelMotor2;
+    private final WPI_TalonFX turretMotor;
+    private final WPI_TalonFX hoodMotor;
+
+    // Constants
+    public static final double FLYWHEEL_VELOCITY = 10;
+    public static final double STOP_MOTOR = 0;
+    public static final double DEGREES_TO_ENCODER_TICKS = 2048 / 360; // 2048 ticks per 360 degrees
+    public static final double MIN_TURRET_ANGLE = -180; // Total ~360 degrees of rotation, assumes 0 is center
+    public static final double MAX_TURRET_ANGLE = 180;
+    public static final double TURRET_MAX_SPEED = 0.1;
+    public static final double TURRET_MIN_SPEED = -0.1;
+    public static final double TURRET_P = 0.01; // Placeholder PID constants
+    public static final double TURRET_I = 0;
+    public static final double TURRET_D = 0;
+    public static final double MAX_HOOD_ANGLE = 40.0;
+    public static final double MIN_HOOD_ANGLE = 5;
+
+    /**
+     * Constructor for shooter subsystem.
+     * 
+     * @param flywheelMotor1
+     *            The first motor connected to the flywheel
+     * 
+     * @param flywheelMotor2
+     *            The second motor connected to the flywheel
+     * 
+     * @param turretMotor
+     *            The motor that controls the horizontal rotation of the
+     *            turret
+     * 
+     * @param hoodMotor
+     *            The motor that controls the angle of the hood above the
+     *            turret
+     * 
+     */
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        // Motor configs
+        var flywheelLimit = new SupplyCurrentLimitConfiguration(true, 40, 40, 500);
+        flywheelMotor1.configSupplyCurrentLimit(flywheelLimit);
+        flywheelMotor1.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.configSupplyCurrentLimit(flywheelLimit);
+        flywheelMotor2.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.setInverted(true);
+
+        var limit = new SupplyCurrentLimitConfiguration(true, 10, 10, 500);
+        TalonFXConfiguration turretConfig = new TalonFXConfiguration();
+        turretConfig.forwardSoftLimitThreshold = MAX_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS;
+        turretConfig.reverseSoftLimitThreshold = MIN_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS;
+        turretConfig.forwardSoftLimitEnable = true;
+        turretConfig.reverseSoftLimitEnable = true;
+        turretMotor.configSupplyCurrentLimit(limit);
+        turretMotor.configAllSettings(turretConfig);
+        turretMotor.setNeutralMode(NeutralMode.Brake);
+        turretMotor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+        turretMotor.config_kP(0, TURRET_P);
+        turretMotor.config_kI(0, TURRET_I);
+        turretMotor.config_kD(0, TURRET_D);
```

Comment:
Can these all be set on `turretConfig` instead of some on `turretConfig` and some directly on the motor?
Also, the configSupplyCurrentLimit() might be overwritten on the next line's `configAllSettings()` call

Reply:
I discovered methods (`configFowardSoftLimitThreshold`, `configReverseSoftLimitEnable`, etc.) that config the soft limits and switched to using those and other motor methods instead of a config object.